aviary/interface/methods_for_level2.py

```diff
@@ -1504,24 +1652,19 @@ def add_linkage_constraint(self, phase_a, phase_b, var_a, var_b, connected,
                 self.model.connect(
                     "traj.ascent.timeseries.altitude", "h_fit.h_cp")
 
-                self.model.connect(
-                    "traj.desc2.timeseries.mass",
-                    "landing.mass",
-                    src_indices=[-1],
-                    flat_src_indices=True,
-                )
+                last_flight_phase_name = list(self.phase_info.keys())[-1]
```

Comment:
Isn't this a lot of steps to just get `self.regular_phases[-1].pathname` ?

Reply:
in 2DOF we actually don't separate regular_phases and normal_phases yet.